src/python_inspector/utils_pypi.py

```diff
@@ -1595,11 +1595,32 @@ def fetch_links(
             url, _, _sha256 = anchor_tag["href"].partition("#sha256=")
             if "data-requires-python" in anchor_tag.attrs:
                 python_requires = anchor_tag.attrs["data-requires-python"]
+            url = resolve_relative_url(package_url, url)  # Resolve relative URL
```

Comment:
Nitpicking: I am not sure the line end comment adds much since this is repeating the function name that is already quite explicit and obvious (and the same as your comment). Also in general we try to avoid end of line comments
```suggestion
            url = resolve_relative_url(package_url, url)
```